ui/src/app/utils/index.tsx

```diff
@@ -527,3 +529,9 @@ export function renderUSD(value: number) {
   }
 }
 
+export const useDebounce = (fn: Function, updateList = [], ms = 250) => {
```

Comment:
OK. How did this bit relate to the rest of the PR?

Reply:
Take a look at [this line](https://github.com/all-of-us/workbench/pull/3736/files#diff-c0c786d256d2a65f6fd35f8f9f60b42cR76).

Debouncing is a generally useful pattern, so rather than coding it up in the useEffect and leaving it up to the developer to figure out what the effect is doing (or adding comments), I created a custom hook `useDebounce`. Since this is a generally useful thing I put it in our utils file. This can be reused by any component that wants to utilize debounce functionality.

Reply:
yeah, a debouncing input box would be nice.